File: 2760. Longest Even Odd Subarray With Threshold/solution.cpp
class Solution {
public:
  int longestAlternatingSubarray(vector<int>& nums, int threshold) {
    int result = 0;
    int last = INT_MAX;

    int index = 0;
    while (index < nums.size()){
      if (nums[index] > threshold || (index > 0 && nums[index] % 2 == nums[index - 1] % 2)){
        if (result < index - last){
          result = index - last;
        }
        last = INT_MAX;
      }
      if (last == INT_MAX && nums[index] <= threshold && nums[index] % 2 == 0){
        last = index;
      }
      index += 1;
    }
    
    if (result < index - last){
      result = index - last;
    }

    return result;
  }
};

File: 2760. Longest Even Odd Subarray With Threshold/solution.py
class Solution(object):
  def longestAlternatingSubarray(self, nums, threshold):
    """
    :type nums: List[int]
    :type threshold: int
    :rtype: int
    """
    result = 0
    last = float('inf')

    index = 0
    while index < len(nums):
      if nums[index] > threshold or (index > 0 and nums[index] % 2 == nums[index - 1] % 2):
        if result < index - last:
          result = index - last
        last = float('inf')
      if last == float('inf') and nums[index] <= threshold and nums[index] % 2 == 0:
        last = index
      index += 1
    
    if result < index - last:
      result = index - last

    return result